tasmota/tasmota_xnrg_energy/xnrg_25_mk_sky_blu.ino

```diff
@@ -0,0 +1,783 @@
+/*
+  xnrg_25_mk_sky_blu.ino - MakeSkyBlue Solar charger support for Tasmota
+
+  Copyright (C) 2024  meMorizE
+
+  This program is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#ifdef USE_ENERGY_SENSOR
+#ifdef USE_MAKE_SKY_BLUE
+/*********************************************************************************************\
+ * This implementation communicates with solar charge controller of MakeSkyBlue(Shenzen) Co.LTD 
+ * http://www.makeskyblue.com
+ * Model: S3-30A/40A/50A/60A MPPT 12V/24V/36V/48V with firmwware V118 (or V119 including a Wifi Box)
+ * The device should have a Mini-USB socket at bottom right beside the screw terminals.
+ * This socket conforms NOT to the USB standard protocol !
+ * It is a TTL serial interface with 9600bps 8N1, D+=Tx D-=Rx plus regular 5V out and GND
+ * When using a DIY ESP hardware be aware of the 5V levels by using e.g. level shifter 
+ * 
+ * The orignal V119 Wifi Box hardware includes an ESP8285 with 1MB Flash (as module 2AL3B ESP-M)
+ * It specific hardware uses
+ *  GPIO1: TX0,  ESP => Charge controller
+ *  GPIO3: RXD0, ESP <= Charge controller
+ *  GPIO5 red LED, active low, for e.g. LedLink_i
+ *  free: GPIO4, GPIO12, GPIO13, GPIO14, GPIO15, GPIO16, GPIO17(ADC)
+ *
+ * Useful: command VoltRes 1 to select voltage resolution to 0.1 V (native resolution)
+ *         command AmpRes 1 to select current resolution to 0.1 A (native resolution)
+ *         SetOption72 to read total energy from the charge controller
+ * 
+ * Note: SetOption129 should be off, because only the solar energy is typical the main interest
+ * 
+ * This implementation is based on own logic analyzer recordings and information from
+ * https://github.com/lasitha-sparrow/Makeskyblue_wifi_Controller
+ * https://www.genetrysolar.com/community/forum-134/topic-1219/
+ * 
+\*********************************************************************************************/
+
+#include <TasmotaSerial.h>
+
+#define XNRG_25                   25
+
+/* compile options */
+#define _VARIANT_ENERGY_IMPORT    2 /* 0=never import, 1=import once, 2=import changed full kWh */
```

Comment:
Please document what these defines do.

Move them to `my_user_config.h` if it makes sense. It will ease configuration